### ISSUE TITLE
No dedicated yasnippet menu when using enh-ruby-mode

### ISSUE THREAD

**Issue by Bounga** [joaotavora/yasnippet#748]
Hi, I'm pretty sure it was working with a previous emacs installation and the same config file but as of today when I'm using enh-ruby-mode, there's no menu (nor completion available) in yasnippet.

I check the package installed by the package manager and enh-ruby-mode is a symlink to ruby-mode. Switching to ruby-mode raise the according yasnippet menu.

I really don't how to find a way to activate ruby-mode snippets when I'm in enh-ruby-mode.

I'm using GNU Emacs 25.1.1, yasnippet 0.11.0 and yas-global-mode is activated.

Do you have any clue about this? Do you need more info?

**Comment by npostavs** [comment]
The only way yasnippet can know that `ruby-mode` snippets should activate for `enh-ruby-mode` is if `ruby-mode` is a parent mode of `enh-ruby-mode`. You could add `.yas-parents` file with contents `ruby-mode` under `enh-ruby-mode` in your snippets directory.

> I check the package installed by the package manager and enh-ruby-mode is a symlink to ruby-mode.

I don't understand what this means. What is a "symlink" in this context? Alias?

**Comment by Bounga** [comment]
It's a Unix file system symbolic link. So enh-ruby-mode directory points to ruby-mode directory. See https://github.com/AndreaCrotti/yasnippet-snippets/blob/29da8e6c0444188406fdc5bb8b18e9b328fb710e/enh-ruby-mode.

It just point to ruby-mode directory and I think it's the trick to make yasnippet load ruby-mode snippets when using enh-ruby-mode.

I never used .yas-parents or anything before and it was working. I'm not sure adding the .yas-parents file by myself in the yasnippet package tree is a good idea since it's handle by the Emacs package manager and it will be lost on the next upgrade.

**Comment by npostavs** [comment]
Oh, you were talking about the files, ok.

I'm not able to reproduce the problem here (yasnippet 8412d71e44576c6e9f182109219a8a71cf9d4311, emacs 25.1, GNU/Linux). 
Can you reproduce this from [`emacs -Q`][emacs-Q]?

[emacs-Q]: http://www.lunaryorn.com/2015/11/29/reproduce-bugs-in-emacs-Q.html

**Comment by Bounga** [comment]
Hi @npostavs it does work with `emacs -Q` so it must be related to my config… Going to disable my config files one by one.

Sorry for the inconvenience.

**Comment by Bounga** [comment]
I tested it again and in fact it doesn't work for me.

Started with `emacs -Q`, then loaded MELPA: 

```
(require 'package)
  (add-to-list
   'package-archives
   '("melpa" . "http://melpa.org/packages/")
   t)
  (package-initialize)
```

then loaded yasnippet: 

```
(require 'yasnippet)
(yas-global-mode 1)
```

It works for ruby-mode but not enh-ruby-mode.

**Comment by npostavs** [comment]
Aha, I can reproduce this when I install via MELPA.

> It's a Unix file system symbolic link. So enh-ruby-mode directory points to ruby-mode directory. See https://github.com/AndreaCrotti/yasnippet-snippets/blob/29da8e6c0444188406fdc5bb8b18e9b328fb710e/enh-ruby-mode.

Actually, when it's installed by MELPA it seems that it's *not* a symlink, I see only an empty file. It looks like it could be a bug in Emacs' tar extraction code, as the tarball itself does have a symlink.

You can create a file .yas-parents file in `~/.emacs/snippets/enh-ruby-mode`, this won't get overwritten by package updates. I will make a PR to do the same in yasnippet-snippets repository so this should get fixed at least by yasnippet 0.12

**Comment by Bounga** [comment]
Ah nice to know that's it's not an issue with my installation. I'm going to use your workaround while waiting for an official fix.

By the way, enh-ruby-mode is not the only mode using this trick in yasnippet-snippets.

Thanks for your help!
